docs/synapse/userguides/ug012_storm_ref_datamod.rst

```diff
@@ -0,0 +1,299 @@
+* Synapse will set the secondary propert(ies) for all nodes returned by ``<query>`` for which that secondary property is a valid property. Nodes for which that property is not a valid secondary property will be ignored.
+* ``setprop()`` will create and set the property if it does not exist, or overwrite the existing ``<prop>=<pval>`` if it does exist.
+* ``setprop()`` can set or modify any property not explicitly defined as read only (``'ro' : 1``) in the data model. Attempts to modify read only properties will fail silently (e.g., the property value will not be overwritten, but the user will not be notified that the request failed).
+* ``setprop()`` cannot be used to remove (delete) a property entirely.
+
+**Operator Syntax Notes:**
+
+* N/A
+
+**Macro Syntax Notes:**
+
+* Synapse will attempt to set the specified propert(ies) for all previously referenced nodes (e.g., to the left of the ``<prop>=<pval>`` statement) for which that property is valid, **whether those nodes are within or outside of the macro syntax brackets.** See `Special Note on Macro Syntax`_.
+
+addtag()
+--------
+
+Adds one or more tags to the specified node(s).
+
+**Operator Syntax:**
+
+.. parsed-literal::
+  *<query>* **addtag(** *<tag>* [ **,** ... ] **)**
+
+**Macro Syntax:**
+
+.. parsed-literal::
+  *<query>* **[** **#** *<tag>* ... **]**
+
+**Examples:**
+
+``inet:fqdn = woot.com addtag( foo.bar , baz.faz )``
+
+``inet:fqdn = woot.com [ #foo.bar #baz.faz ]``
+
+**Usage Notes:**
+
+* ``addtag()`` operates on the output of a previous Storm query.
+* Synapse will apply the specified tag(s) to all nodes returned by ``<query>``.
+
+**Operator Syntax Notes:**
+
+* N/A
+
+**Macro Syntax Notes:**
+
+* Synapse will set the specified tag(s) for all previously referenced nodes (e.g., to the left of the ``<tag>`` statement) **whether those nodes are within or outside of the macro syntax brackets.** See `Special Note on Macro Syntax`_.
+
+delnode()
+---------
+
+Deletes the specified node(s) from a Cortex.
+
+**Operator Syntax:**
+
+.. parsed-literal::
+  *<query>* **delnode(force=1)**
+
+**Macro Syntax:**
+
+None.
+
+**Examples:**
+
+``inet:fqdn = woot.com delnode(force=1)``
+
+**Usage Notes:**
+
+* ``delnode()`` operates on the output of a previous Storm query.
+* Use of the ``force=1`` parameter is meant to require the user to think about what they're doing before executing the ``delnode()`` command (there is no "are you sure?" prompt). Future releases of Synapse will support a permissions structure that will limit the users who are able to execute this operator.
+* ``delnode()`` has the potential to be destructive if executed on an incorrect, badly formed, or mistyped query. Users are strongly encouraged to validate their query by first executing it on its own to confirm it returns the expected nodes before executing ``delnode()``. (Picture accidentally running ``inet:fqdn delnode( force=1 )`` instead of ``inet:fqdn = woot.com delnode( force=1 )``.)
```

Comment:
Might want to tell them what that Picture means :) in case they can't follow the syntax quite yet.  :D